### ISSUE TITLE
Model::hasChanged($params) always return true if $params is array and any field is changed

### ISSUE THREAD

**Issue by mzf** [phalcon/cphalcon#12669]
### Expected and Actual Behavior
Model::hasChanged() always return true if params is array and any field is changed.
I think if send array to model::hasChanged, this method should check that fields in this array has changed.

Something like this

```php
public function hasChanged(var fieldName = null) -> boolean
	{
		var changedFields;

		let changedFields = this->getChangedFields();

		/**
		 * If a field was specified we only check it
		 */
		if typeof fieldName == "string" {
			return in_array(fieldName, changedFields);
		} elseif (typeof fieldName == "array") {
                       foreach (changedFields  as field) {
                            if in_array(field, changedFields) {
                                  return true;   
                            }                             
                       } 
                       return false;   
                }

		return count(changedFields) > 0;
	}
```

* Phalcon version: 3.0.4
* PHP Version: 7.0.15
* Operating System: Win7

**Comment by Jurigag** [comment]
I don't like this, maybe someone would like to check if only one of parameter of array was changed?

https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model.zep#L3824

Actually if array is provided nothing happens with it right now.

I would like to add option strict, if false(by default?) it will do what you want, if true it will check all keys, like array_intersect.

**Comment by Jurigag** [comment]
Created some PR for this, let me know what you think.

**Comment by mzf** [comment]
This is what I needed. Thanks!

**Comment by sergeyklay** [comment]
Fixed in the 3.2.x branch
